### ISSUE TITLE
complex_plot not showing in Chrome (works in Firefox & Safari)

### ISSUE THREAD

**Issue by gabindu** [sagemathinc/smc#209]
The following shows an empty plot (with coordinate system) in Chrome (tested in version 46.0.2490.80, 64-bit on OS X 10.11.1), but works fine in Firefox (41.0.2) and Safari (9.0.1):

    complex_plot(lambda z: z, (-3, 3), (-3, 3))

(this is directly pulled from the complex_plot? documentation).
No error is shown, but the (usually colorful) graphics is simply not shown.

Interestingly, saving the plot as a png or pdf **does** produce the correct picture (attached below).

    P=complex_plot(lambda z: z, (-3, 3), (-3, 3))
    P.save("test.png")

![test](https://cloud.githubusercontent.com/assets/12254984/10873996/e38e0e00-80ea-11e5-8391-ae20239be986.png)

Not sure if this is an isolated bug in Chrome, or rather a SMC-specific problem, but I thought I'd raise the issue here anyways.

**Comment by gabindu** [comment]
Even more interesting: opening the saved plot (e.g. test.png, as above) in SMC, is shown correctly even inside Chrome. But it is **not** shown inside the Sage worksheet.

    P=complex_plot(lambda z: z, (-3, 3), (-3, 3))
    P.show()    # shows empty coordinate system
    P.save("test.png")
    salvus.open_tab("test.png")   # shows correctly in new SMC tab

**Comment by williamstein** [comment]
1. In SMC with Chrome a workaround is to do:

   P=complex_plot(lambda z: z, (-3, 3), (-3, 3))
   P.show(svg=False)

https://cloud.sagemath.com/projects/4a5f0542-5873-4eed-a85c-a18c706e8bcd/files/support/2015-11-01-203719-complex.sagews

2. I actually can't replicate this problem in Chrome.  Which OS are you
using?

William

On Sun, Nov 1, 2015 at 8:26 PM, Gabriel Indurskis <notifications@github.com>
wrote:

> Even more interesting: opening the saved plot (e.g. test.png, as above) in
> SMC, is shown correctly even inside Chrome. But it is *not* shown inside
> the Sage worksheet.
>
> P=complex_plot(lambda z: z, (-3, 3), (-3, 3))
> P.show()    # shows empty coordinate system
> P.save("test.png")
> salvus.open_tab("test.png")   # shows correctly in new SMC tab
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sagemathinc/smc/issues/209#issuecomment-152909683>.
>



-- 
William (http://wstein.org)

**Comment by gabindu** [comment]
ok, this is rather mysterious: I opened your worksheet and everything displayed fine. Reopening my own worksheet, it showed the empty coordinate system again. I reloaded (and shift-reloaded), same problem. I created a new worksheet with the same command, same problem. I closed the browser tab, and reloaded SMC in a different tab - suddenly all the worksheets show the correct output.

This is on OS X El Capitan (10.11.1), Chrome 46.0.2490.80.
![screen shot 2015-11-02 at 12 18 00 am](https://cloud.githubusercontent.com/assets/12254984/10874744/7849012e-80f7-11e5-9b34-8b0e2c26ce2f.png)
![screen shot 2015-11-02 at 12 18 56 am](https://cloud.githubusercontent.com/assets/12254984/10874745/78490908-80f7-11e5-8e95-f110c168a0d6.png)
